### DISCUSSION TITLE
MIN{} and MAX{} aggregations not returning correct data for full record

### DISCUSSION THREAD

**Discussion by ShivaniSK** [finos/TimeBase-CE#294] [Q&A]
Hi @alex-karpovich 

We have data in timebase with below format.
We want to get the max value of price and corresponding timestamp with venue for that price value.

Price       Timestamp                               Venue
151.775   2024-04-10T00:00:51.940Z      Reuters
151.715   2024-04-10T00:00:57.048Z      EBS

On running below query
```
select MAX{} (price) as priceAgg, timestamp as timestampOg, venue as venueOg  
                from market_trades OVER TIME (1m) 
            WHERE timestamp BETWEEN 1712707200000 AND 1712793599999 
AND SYMBOL == 'USDJPY' 
```
OR
```
select MAX{} (price) as priceAgg, MAX{} (timestamp) as timestampOg, MAX{} (venue) as venueOg  
                from market_trades OVER TIME (1m) 
            WHERE timestamp BETWEEN 1712707200000 AND 1712793599999 
AND SYMBOL == 'USDJPY' 
```

The expected output is the record having max price value and the corresponding timestamp and venue
As per above snip, the expected output should be

**Expected output**
Price       Timestamp                               Venue
151.775   2024-04-10T00:00:51.940Z      Reuters

**Actual output**
Price       Timestamp                               Venue
151.775   2024-04-10T00:00:57.048Z      EBS

The same happens on running below query where it gives the min price value correct, but timestamp and venue are incorrect.
```
select MIN{} (price) as priceAgg, timestamp as timestampOg, venue as venueOg  
                from market_trades OVER TIME (1m) 
            WHERE timestamp BETWEEN 1712707200000 AND 1712793599999 
AND SYMBOL == 'USDJPY' 
```

**Comment by alex-karpovich** [comment]
Greetings, I assume you need to add `GROUP BY venue ` in your query

**Reply by ShivaniSK** [reply]
Hi @alex-karpovich , 
Group by venue won't help, as we do not want to have record for each venue.
Our requirement is to have 1 record per minute, where it should pick the record which has MAX price value. The timestamp and venue should be for that corresponding max price value record only. 

So if the dataset is like below

Price       Timestamp Venue
151.765   2024-04-10T00:00:51.940Z Reuters
151.715   2024-04-10T00:00:57.048Z EBS
**151.775   2024-04-10T00:00:57.900Z Reuters**
151.711   2024-04-10T00:00:58.048Z EBS

We want to get the record with MAX price value with corresponding timestamp and venue
Expected output: **151.775   2024-04-10T00:00:57.900Z Reuters**
Actual output: 151.775  **2024-04-10T00:00:58.048Z EBS**

Can you please suggest the TimeBase query for the above use-case.

Regards,
Shivani

**Comment by alex-karpovich** [comment]
@ShivaniSK when you are using any aggregate function in query, all other columns without aggregates automatically inherit `LAST` aggregate function, so your query is same as 
```
select MIN{} (price) as priceAgg, LAST{}(timestamp) as timestampOg, LAST{}(venue) as venueOg  
                from market_trades OVER TIME (1m) 
            WHERE timestamp BETWEEN 1712707200000 AND 1712793599999 
AND SYMBOL == 'USDJPY' 
```

So I can suggest to use `GROUP BY venue` and filter results after query on java/python side
